scripts/orb-registration-script/orb-registration.py

```diff
@@ -0,0 +1,754 @@
+        return pairs
+
+    def run(self):
+        """Main execution logic."""
+        cf_token = self.get_cloudflared_token()
+
+        if self.args.platform == "pearl":
+            # Pearl: Generate artifacts and register
+            self.build_dir.mkdir(exist_ok=True)
+            self.artifacts_dir.mkdir(exist_ok=True)
+
+            with tempfile.TemporaryDirectory() as temp_dir:
+                mount_point = Path(temp_dir) / "loop"
+                mount_point.mkdir()
+
+                self.create_persistent_images(mount_point)
+
+                for i in range(self.args.count):
+                    self.logger.info(
+                        f"Generating Pearl Orb ID #{i+1} of {self.args.count}..."
+                    )
+                    orb_id = self.process_pearl_orb(cf_token, mount_point)
+                    self.logger.info(f"Successfully processed Pearl Orb: {orb_id}")
+                    print("", file=sys.stderr)
+
+                self.logger.info(
+                    f"All {self.args.count} Pearl Orb IDs generated and registered successfully."
+                )
+
+        elif self.args.platform == "diamond":
+            if self.args.input_file:
+                # Diamond: Read from file
+                if self.args.input_format == "ids":
+                    orb_ids = self.read_input_file(self.args.input_file)
+                    self.process_diamond_orb_ids(orb_ids, cf_token)
+                elif self.args.input_format == "pairs":
+                    orb_pairs = self.read_input_pairs_file(self.args.input_file)
+                    self.process_diamond_orb_pairs(orb_pairs)
+            elif self.args.orb_ids:
+                # Diamond: Direct arguments
+                self.process_diamond_orb_ids(self.args.orb_ids, cf_token)
+            else:
+                raise ValueError(
+                    "Diamond platform requires either --input-file or direct orb IDs"
+                )
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Generate and register Orb IDs for both Pearl and Diamond platforms"
+    )
+
+    parser.add_argument(
+        "--platform",
+        choices=["pearl", "diamond"],
+        required=True,
+        help="Platform type (pearl or diamond)",
+    )
+    parser.add_argument(
+        "--backend",
+        choices=["stage", "prod"],
+        required=True,
+        help="Backend environment",
+    )
+    parser.add_argument(
+        "--release", choices=["dev", "prod"], required=True, help="Release type"
+    )
+    parser.add_argument(
+        "--hardware-version",
+        required=True,
+        help="Hardware version (e.g., PEARL_EVT1, DIAMOND_EVT2)",
+    )
+    parser.add_argument(
+        "--mongo-token", help="MongoDB bearer token (overrides environment)"
+    )
+    parser.add_argument(
+        "--core-token", help="Core-App bearer token (overrides environment)"
+    )
+    parser.add_argument(
+        "--channel",
+        default="general",
```

Comment:
for prod, let's not set a default channel and hence if channel not set don't set it when you process orb ids.

In theory we could set different channels based on platform, but I think we first need to clean them up